csharp/sample/InferenceSample/Microsoft.ML.OnnxRuntime.InferenceSample.Forms/MainPage.xaml.cs

```diff
@@ -18,34 +21,79 @@ protected override void OnAppearing()
         {
             base.OnAppearing();
 
-            Console.WriteLine("Using API");
-            inferenceSampleApi.Execute();
-            Console.WriteLine("Done");
+            OutputLabel.Text = "Press 'Run Tests'.\n";
+        }
+
+        private readonly InferenceSampleApi inferenceSampleApi;
+
+        private async Task ExecuteTests() 
+        {
+            Action<Label, string> addOutput = (label, text) =>
+            {
+                Device.BeginInvokeOnMainThread(() => { label.Text += text; });
+                Console.Write(text);
+            };
+
+            OutputLabel.Text = "Testing execution\nComplete output is written to Console in this trivial example.\n\n";
+
+            await Task.Run(() =>
```

Comment:
why do we need to put the logic in multiple `Task.Run()`s?

Reply:
Good point - a single task is fine. Updated.